security/pkg/k8s/controller/workloadsecret.go

```diff
@@ -132,6 +138,21 @@ type SecretController struct {
 	// Used to coordinate with label and check if this instance of Citadel should create secret
 	istioCaStorageNamespace string
 	monitoring              monitoringMetrics
+
+	// Extra trust anchors related state
+	extraTrustAnchorsController cache.Controller
+	extraTrustAnchorMu          sync.Mutex
+	// extra trust anchors organized by source configmap name and key within that configmap.
+	// The value is a PEM encoded root certificate. Extra anchors are combined with the
+	// primary root to form the list of trusted roots distributed to proxies.
+	extraTrustAnchors map[string]map[string]string // configmap name -> configmap key -> PEM encoded cert
```

Comment:
This seems first time we introduce this terminolgoy in our code base(maybe in the doc as well?)

- Can't we just name it as `meshRootBundle` or something refers to root?
- The use case is one config map corresponds to one cluster's root? And it contains multiple root? Why we need to have multi root in a single config map? What are the key in the config map stands for?

@myidpt wdty?

Reply:
I'll defer to @myidpt on naming but extra_trust_* seems reasonable to me.

User's have some flexibility in how they manage additional certs. For the multi-cluster 'join' case I'll probably use one-root-per-configmap with the configmap and key name corresponding to the name of the remote cluster. Other users might bundle multiple additional trusted roots in a single configmap. The only requirement for the key is that it is unique per configmap which is already enforced by k8s. The citadel dedups the full list of roots to minimize the bundle size distributed to workloads.

Reply:
> For the multi-cluster 'join' case I'll probably use one-root-per-configmap with the configmap and key name corresponding to the name of the remote cluster.

Sorry I'm confused. One root per config map, but key name corresponds to the remote cluster name (we can have multiple remote cluster), how could we make it only one root?

We have multiple cluster, each of them can have different root, but this is just one level multiplicity, not sure why we need both multi config map and multi entries in each config map.

Reply:
Users can provide one or more roots in each configmap. The configmap name and key(s) are opaque to Citadel. Only the field _value_ matters (i.e. a PEM encoded root cert). Users are free to use their own naming convention. Citadel merges the root(s) from all configmaps along it's own root to form the complete list of roots.

For multi-cluster, I'm proposing the convention be one-remote-cluster-root per configmap with the name and key both corresponding to the remote cluster's ID/name. This makes it easy for tooling to manage the list of roots by creating/deleting resources and avoids the need to patch an existing resource. Other non-multi-cluster use cases can easily co-exist with this scheme. 

multiple-roots-per-configmap isn't required but its easy to support and understand IMO, i.e. any root(s) in a configmap labeled accordingly will be appended to the list of roots. 

To restrict usage to one-root-per-configmap requires using a known key (e.g. default) which is one more config to think about and check.

Reply:
I personally prefer not adding API/hooks to allow multi roots within a single mesh unless we have a strong use case requiring that.

Also note, based on description, we are now planning to implicitly, support `cluster.local` same trust domain and using different roots in each cluster. Is this today's single mesh multi cluster setup? Or it's something of status quo?

Just want to call out. @myidpt @smawson @nmittler

Reply:
The use case is to establish initial trust between existing clusters within minimal disruption. I expect future work will enable establishing a common CA between clusters (e.g. key rotation?). The alternative for the self-signed cert case is to copy private keys between clusters and experience downtime during rotation.